locksmithctl/daemon.go

```diff
@@ -147,6 +195,9 @@ func (r rebooter) lockAndReboot(lck *lock.Lock) {
 
 			continue
 		}
+		if err := getPeriodicAndSleep(); err != nil {
```

Comment:
now we have this function, we could just move it above, into the `if err != nil && err != lock.ErrExist` block.

We start the `lockAndReboot`; if `lck.Lock()` fails to acquire a semaphore, we `time.Sleep(interval)`; then we call `getPeriodicAndSleep`: this one will ensure we are still inside the reboot window because of this lines:
```go
		if sleeptime > 0 {
			dlog.Infof("Reboot window start is %q and length is %q", startw, lengthw)
			next := period.Next(time.Now())
			dlog.Infof("Next window begins at %s and ends at %s", next.Start, next.End)
			dlog.Infof("Waiting for %s to reboot.", sleeptime)
			time.Sleep(sleeptime)
		}
```
so if `sleeptime > 0`, we sleep until being into the reboot window else we fall into the `continue` so we loop and try to acquire a semaphore and so on on.

The current position of the call of this function would imply to keep the semaphore until we reach again the reboot window - which could prevent other instance to reboot.